src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -129,19 +144,64 @@ private void addVariableIfNotNullAndModelIfNotEmpty(
         }
         if (checkIfNotNull(name, model) && checkIfNotEmpty(name, model)) {
             keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(name, model, ATTRIBUTE_VARIABLE_MODEL_LABEL));
+            addDependsOn(keyValueMapBuilder, value, model);
+        }
+    }
+
+    private void addDependsOnIfCatalogObjectModelExistOnTaskVariable(
+            ImmutableSet.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, TaskVariable taskVariable) {
+        String name = taskVariable.getName();
+        String value = taskVariable.getValue();
+        String model = taskVariable.getModel();
+        if (checkIfNotNull(name, model) && checkIfNotEmpty(name, model)) {
+            addDependsOn(keyValueMapBuilder, value, model);
+        }
+
+    }
+
+    private void addDependsOn(ImmutableSet.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, String value,
+            String model) {
+        if (model.equalsIgnoreCase(CATALOG_OBJECT_MODEL)) {
+            if (getRevisionFromDependencyOn(value).isPresent()) {
+                keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(getNameAndBucketFromDependencyOn(value),
+                                                                       getRevisionFromDependencyOn(value).get(),
+                                                                       ATTRIBUTE_DEPENDS_ON_LABEL));
+            } else {
+                keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(getNameAndBucketFromDependencyOn(value),
+                                                                       LATEST_VERSION,
+                                                                       ATTRIBUTE_DEPENDS_ON_LABEL));
+            }
+        }
+    }
+
+    private String getNameAndBucketFromDependencyOn(String calledWorkflow) {
+        try {
+            return calledWorkflow.split("/")[0] + "/" + calledWorkflow.split("/")[1];
```

Comment:
please do a single split call, not a big deal, but it's prettier

Reply:
I agree